packages/adapter-kysely/src/index.ts

```diff
@@ -0,0 +1,435 @@
+/**
+ * <div style={{display: "flex", justifyContent: "space-between", alignItems: "center", padding: 16}}>
+ *  <p style={{fontWeight: "normal"}}><a href="https://koskimas.github.io/kysely/">Kysely</a> adapter for Auth.js / NextAuth.js.</p>
+ *  <a href="https://koskimas.github.io/kysely/">
+ *   <img style={{display: "block"}} src="https://raw.githubusercontent.com/nextauthjs/adapters/main/packages/kysely/logo.svg" width="38" />
+ *  </a>
+ * </div>
+ *
+ * ## Installation
+ * ```bash npm2yarn2pnpm
+ * npm install next-auth kysely pg @next-auth/kysely-adapter
+ * npm install --save-dev @types/pg
+ * ```
+ *
+ * @module @next-auth/kysely-adapter
+ */
+import { Kysely, SqliteAdapter } from "kysely"
+import type { Adapter } from "next-auth/adapters"
+import type { Database } from "./database"
+
+type ReturnData<T = never> = Record<string, Date | string | T>
+/**
+ * ## Basic usage
+ * This is the Kysely Adapter for [`next-auth`](https://authjs.dev). This package can only be used in conjunction with the primary `next-auth` package. It is not a standalone package.
+ *
+ * This adapter supports the same first party dialects that Kysely (as of v0.23.5) supports: PostgreSQL, MySQL, and SQLite. The examples below use PostgreSQL with the [pg](https://www.npmjs.com/package/pg) client.
+ * ### Configure Auth.js
+ * ```typescript title="pages/api/auth/[...nextauth].ts"
+ * import NextAuth from "next-auth"
+ * import GoogleProvider from "next-auth/providers/google"
+ * import { KyselyAdapter } from "@next-auth/kysely-adapter"
+ * import { db } from "../../../db"
+ *
+ * export default NextAuth({
+ *   adapter: KyselyAdapter(db),
+ *   providers: [
+ *     GoogleProvider({
+ *       clientId: process.env.GOOGLE_CLIENT_ID,
+ *       clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+ *     }),
+ *   ],
+ * })
+ * ```
+ * ### Configure Kysely
+ * Kysely expects you to define interfaces for each of your database tables before passing them to the `Kysely` constructor.
+ *
+ * This adapter also exports a wrapper around the original Kysely class, `AuthedKysely`, that can be used to provide an additional level of type safety. While using it isn't required, it is recommended as it will verify that the database interface has all the fields that Auth.js requires.
```

Comment:
```suggestion
 * This adapter also exports a wrapper around the original Kysely class, `AuthedKysely`, that can be used to provide an additional level of type-safety. While using it isn't required, it is recommended as it will verify that the database interface has all the fields that Auth.js requires.
```